packages/server/lib/services/connectSession.service.ts

```diff
@@ -104,6 +105,19 @@ export async function getConnectSession(
     return Ok(ConnectSessionMapper.from(session));
 }
 
+export async function getConnectSessionByToken(db: knex.Knex, token: string): Promise<Result<ConnectSession, ConnectSessionError>> {
+    const getSession = await keystore.getPrivateKey(db, token);
+    if (getSession.isErr()) {
+        return Err(new ConnectSessionError({ code: 'not_found', message: `Token not found`, payload: { token: `${token.substring(0, 32)}...` } }));
+    }
+    const privateKey = getSession.value;
+    const session = await getConnectSession(db, { id: privateKey.entityId, accountId: privateKey.accountId, environmentId: privateKey.environmentId });
+    if (session.isErr()) {
+        return Err(session.error);
+    }
+    return Ok(session.value);
```

Comment:
you can't return session directly?

Reply:
why not?

Reply:
I meant this: 
```suggestion
    return session
```

instead of the current if err ok

Reply:
ok yes. you are right. the unwrapping/wrapping is unnecessary. I will fix next time I touch the code